### PR TITLE
Remove user and bookmarks, move to isomorphic fetch

### PR DESCRIPTION
**Pull request by mmahalwy**
### Title of change
- Removes user & bookmarks from redux connects
- Replaces superagent with isomorphic-fetch
- Adds fetch mock for jest
- Fixes tests
- Changes how ApiClient was setup

As written to @naveed-ahmad on slack:

I am trying to thin down the files (by removing the auth and bookmark stuff) and rewrite some things in a better way. I learnt a lot at Airbnb so far and want to bring those learnings to the project. Hence, removing unused code for the time being. Until we focus on one.quran.com again. But we had a lot of dead code :disappointed:

### Checklist

- [x] Unit tests written
- [x] Manually tested
- [x] Prettier & ESLint were run
- [x] New dependencies are included in `package-lock.json`

### Screenshot
_insert screenshot if needed_

### CODE REVIEW

**Comment by mmahalwy** [comment]
rebuild

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32921

**Comment by mmahalwy** [comment]
rebuild

**Comment by mmahalwy** [comment]
```docker: Error response from daemon: Conflict. The container name "/github_pr920" is already in use by container "98ad010011ab8149d0ab9a31cad4fe11f602887ea50932134b3dddd448262b43". You have to remove (or rename) that container to be able to reuse that name.
```

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32922

**Comment by mmahalwy** [comment]
@ahmedre any way i can see the logs for last build? Just says internal server error

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32923

**Comment by naveed-ahmad** [comment]
@mmahalwy 

ReferenceError: fetch is not defined     at ApiClient.get (/quran/dist/helpers/ApiClient.js:82:21)     at loadAll (/quran/dist/redux/actions/chapters.js:34:21)     at chaptersConnect (/quran/dist/containers/Surah/connect.js:74:41)     at /quran/dist/server.js:173:23     at Array.forEach (native)     at /quran/dist/server.js:172:22     at Array.some (native)     at /quran/dist/server.js:168:18     at Layer.handle [as handle_request] (/quran/node_modules/express/lib/router/layer.js:95:5)     at trim_prefix (/quran/node_modules/express/lib/router/index.js:312:13)     at /quran/node_modules/express/lib/router/index.js:280:7     at Function.process_params (/quran/node_modules/express/lib/router/index.js:330:12)     at next (/quran/node_modules/express/lib/router/index.js:271:10)     at bound (domain.js:282:14)     at runBound (domain.js:293:12)     at Raven.extend.context (/quran/node_modules/raven/lib/client.js:267:40)     at /quran/node_modules/raven/lib/client.js:396:12     at Layer.handle [as handle_request] (/quran/node_modules/express/lib/router/layer.js:95:5)     at trim_prefix (/quran/node_modules/express/lib/router/index.js:312:13)     at /quran/node_modules/express/lib/router/index.js:280:7     at Function.process_params (/quran/node_modules/express/lib/router/index.js:330:12)     at next (/quran/node_modules/express/lib/router/index.js:271:10)     at SendStream.error (/quran/node_modules/serve-static/index.js:121:7)     at emitOne (events.js:96:13)     at SendStream.emit (events.js:188:7)     at SendStream.error (/quran/node_modules/serve-static/node_modules/send/index.js:275:17)     at SendStream.onStatError (/quran/node_modules/serve-static/node_modules/send/index.js:392:12)     at next (/quran/node_modules/serve-static/node_modules/send/index.js:711:28)     at /quran/node_modules/serve-static/node_modules/send/index.js:719:23     at FSReqWrap.oncomplete (fs.js:123:15)
--

**Comment by naveed-ahmad** [comment]
Exception are recorded at:  https://sentry.io/qurancom/web/

File: src/customPropTypes.js [hunk 1]
```diff
@@ -121,9 +121,9 @@ export const surahType = PropTypes.shape({
 export const juzType = PropTypes.shape({
   id: PropTypes.number.isRequired,
   juzNumber: PropTypes.number.isRequired,
-  nameArabic: PropTypes.string.isRequired,
-  nameSimple: PropTypes.string.isRequired,
-  verseMapping: PropTypes.arrayOf(PropTypes.object)
+  nameArabic: PropTypes.string,
+  nameSimple: PropTypes.string,
+  verseMapping: PropTypes.object
```
Comment (naveed-ahmad) [inline src/customPropTypes.js]: `verseMapping` should be required attribute, and we can remove `nameArabic ` and  `nameSimple `.

File: .eslintrc [hunk 2]
```diff
@@ -16,14 +16,11 @@
     "classes": true,
     "jsx": true
   },
-  "plugins": [
-    "react",
-    "mocha"
-  ],
+  "plugins": ["react", "mocha"],
```
Comment (sabeurthabti) [inline .eslintrc]: arent we on jest ? lets remove mocha plugin

**Comment by mmahalwy** [comment]
Oh man, @naveed-ahmad we got lots of exceptions. Also, we have exceeded our monthly quota... we might have to host sentry ourselves cc @hammady @ahmedre

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32924

**Pull request merged**